extension/oidcextension/factory.go

```diff
@@ -0,0 +1,61 @@
+// Copyright -c Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oidcextension
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/extension/extensionhelper"
+)
+
+const (
+	typeStr = "oidc"
+)
+
+type OIDCFactory struct {
+}
+
+func NewFactory() component.ExtensionFactory {
+	return extensionhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		createExtension)
```

Comment:
Does it provide a `WithStart`, where you could specify that your `Start()` function should be called? Same with `WithShutdown`. If it doesn't, your `createExtension` should then call the `Start()` function before adding to the registry.

Reply:
As you mentioned above, the extension does get started by the collector so adding another call isn't necessary